### PR TITLE
token-js: cpi guard

### PR DESCRIPTION
**Pull request by 2501babe**
part of #3771

### CODE REVIEW

**Comment by 2501babe** [comment]
most this this is just the memo transfer code reworked btw

File: token/js/src/extensions/cpiGuard/instructions.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    account: PublicKey,
+    authority: PublicKey,
+    multiSigners: Signer[] = [],
+    programId = TOKEN_2022_PROGRAM_ID
+): TransactionInstruction {
+    return createCpiGuardInstruction(CpiGuardInstruction.Disable, account, authority, multiSigners, programId);
+}
+
+function createCpiGuardInstruction(
+    enable: CpiGuardInstruction,
```
Comment (joncinque) [inline token/js/src/extensions/cpiGuard/instructions.ts:61]: nit: maybe call this `cpiGuardInstruction`?

File: token/js/src/extensions/memoTransfer/instructions.ts [hunk 2]
```diff
@@ -57,11 +57,11 @@ export function createDisableRequiredMemoTransfersInstruction(
     multiSigners: Signer[] = [],
     programId = TOKEN_2022_PROGRAM_ID
 ): TransactionInstruction {
-    return createMemoTransferInstruction(/* enable */ false, account, authority, multiSigners, programId);
+    return createMemoTransferInstruction(MemoTransferInstruction.Disable, account, authority, multiSigners, programId);
 }
 
 function createMemoTransferInstruction(
-    enable: boolean,
+    enable: MemoTransferInstruction,
```
Comment (joncinque) [inline token/js/src/extensions/memoTransfer/instructions.ts:64]: nit: while we're at it, `memoTransferInstruction`?

**Review by joncinque — APPROVED** [review]
Looks great! Just two little nits

**Pull request merged**